External Renderer/Assets/Scripts/ImportScene.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace SceneStateExporter
+{
+#if UNITY_EDITOR
+    [CustomEditor(typeof(ImportScene))]
+    public class QuickEditorImport : Editor
+    {
+        public override void OnInspectorGUI()
+        {
+            DrawDefaultInspector();
+
+            EditorGUILayout.HelpBox("Click Import Now to Import now.", MessageType.None);
+
+            ImportScene currentImporter = target as ImportScene;
+            if (GUILayout.Button("Import Now"))
+            {
+                currentImporter.ImportCurrentScene();
+            }
+        }
+    }
+#endif
+
+    public class ImportScene : MonoBehaviour
+    {
+        public string ImportFilePath = @"D:\Virtana\obj.json";
+        public string ImageSaveFolder = @"D:\Virtana\Planning";
+
+        // Represents when the scene was exported.
+        // if null, then no import has occured
+        public DateTime? ExportTimestamp;
+
+        public void ImportCurrentScene()
+        {
+            Debug.Log("Beginning Import.");
+            // possibly implement a pause engine state here
+
+            // check if file exists
+
+            // get all current items in scene
+            Scene currentScene = SceneManager.GetActiveScene();
+            List<GameObject> importObjects = new List<GameObject>();
+            currentScene.GetRootGameObjects(importObjects);
+            if (importObjects == null || importObjects.Count == 0)
+            {
+                Debug.LogWarning("Empty object List.");
+                return;
+            }
+
+            foreach (var gObj in importObjects)
+            {
+                gObj.transform.SetParent(transform, true);
+            }
+
+            string json = System.IO.File.ReadAllText(ImportFilePath);
+
+            Debug.Log("Deserializing...");
+            var state = JsonConvert.DeserializeObject<SceneState>(json);
```

Comment:
'var', also this would be a good place to do some error checking, where the string actually gets deserialized (or add a comment reminding us to do it later)

Reply:
Removed var, added basic fail checking with log and TODO for hooking into the Serializer for errors.